syntax = "proto3";  //声明了protobuf的版本

package fixbug; //声明了代码所在的包（对于c++来说是namespace）

//定义下面的选项，表示生成service服务嘞和rpc方法描述 默认是不生成的
option cc_generic_services = true;

message RequestCode{
    int32 errcode = 1;
    bytes errmsg = 2;
}

//数据  映射表
//定义登陆请求消息类型 name pwd
message LoginRequest{
    bytes name = 1;
    bytes pwd = 2;
    map<int32, string> test = 3;   //映射表
}

//定义登陆响应消息类型
message LoginResponse{
    RequestCode result = 1;
    bool success = 2;
}

//列表
message GetFriendListRequest{
    uint32 userid = 1;
}

message User{
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListResponse{
    RequestCode result = 1;
    repeated User friend_list = 2;  //注意这个repeated 定义了一个列表类型
}

//在protobuf中如何定义描述rpc方法的类型 - service
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListRequest) returns(GetFriendListResponse);
}
